docs: Add an initial README outlining instructions for project setup and linting tasks. (#10)
y-scope/ystdlib-cpp: README.md

ystdlib-cpp
===================================
An open-source C++ library developed and used at YScope.

# Contributing
Follow the steps below to develop and contribute to the project.

## Requirements
* Python 3.10 or higher
* [Task] 3.40.0 or higher

## Set up
Initialize and update submodules:
```shell
git submodule update --init --recursive
```

## Linting

> Before submitting a pull request, ensure you’ve run the linting commands below and either fixed any
> violations or suppressed the warnings.

Reviewer: ```suggestion
Before submitting a pull request, ensure you’ve run the linting commands below and have fixed all
violations and suppressed all warnings.
```

To run all linting checks:
```shell
task lint:check
```

> To run all linting checks AND automatically fix any fixable issues:

Reviewer: ```suggestion
To run all linting checks AND fix some violations:
```
Just matching the wording in spider.
